frontend-react/src/hooks/network/ServiceHooks.ts

```diff
@@ -0,0 +1,101 @@
+
+import { auxExports } from "../UseCreateFetch";
+import {
+    MembershipState,
+    MemberType,
+    ServiceSettings,
+} from "../UseOktaMemberships";
+import { RSService, servicesEndpoints } from "../../config/endpoints/settings";
+import { RSNetworkError } from "../../utils/RSNetworkError";
+
+type ServiceState = ServiceSettings & {
+    setActiveService: Dispatch<SetStateAction<string>>;
+};
+/** Fetches membership services (senders, receivers) from the ReportStream API. Updates
+ * whenever the given {@link MembershipState} or AccessToken (Okta) are updated.
+ * @remarks ONLY FOR USE WITHIN {@link SessionContext} */
+export const useMemberServices = (
+    state: MembershipState,
+    token: AccessToken | undefined
+): ServiceState => {
+    const [senders, setSenders] = useState<RSService[] | undefined>();
+    const [receivers, setReceivers] = useState<RSService[] | undefined>();
+    const [activeService, setActiveService] = useState<string>("default");
+    /* Because this is used at the SessionContext > useOktaMemberships level, it does not have access
+     * to the AuthorizedFetchProvider nested within SessionContext. For this, we must generate an auth fetch
+     * manually using our custom tooling. */
+    const authFetchServicesGenerator = useCallback(() => {
+        return auxExports.createTypeWrapperForAuthorizedFetch(token!!, {
+            parsedName: state?.activeMembership?.parsedName || "",
+            memberType:
+                state?.activeMembership?.memberType || MemberType.NON_STAND,
+        });
+    }, [
+        state?.activeMembership?.memberType,
+        state?.activeMembership?.parsedName,
+        token,
+    ]);
+    /* Logic gate for whether we should or should not fetch */
+    const hasAllNecessaryVariablesForFetch = useMemo(
+        () =>
+            state?.initialized &&
+            state?.activeMembership?.memberType &&
+            state?.activeMembership?.parsedName &&
+            token,
+        [
+            state?.activeMembership?.memberType,
+            state?.activeMembership?.parsedName,
+            state?.initialized,
+            token,
+        ]
+    );
+    /* Using async auth fetches to retrieve arrays of sender/receiver services from API
+     * and setting local state */
+    useEffect(() => {
+        const fetcher = authFetchServicesGenerator();
+        const fetcherOptions = {
+            segments: {
+                orgName: state?.activeMembership?.parsedName!!,
+            },
+        };
+        const fetchData = async () => {
+            const { senders, receivers } = servicesEndpoints;
+            const senderServiceResults: RSService[] = await fetcher<
+                RSService[]
+            >(senders, fetcherOptions);
+            const receiverServiceResults: RSService[] = await fetcher<
+                RSService[]
+            >(receivers, fetcherOptions);
+            setSenders(senderServiceResults);
+            setReceivers(receiverServiceResults);
```

Comment:
There's a _lot_ of data in these settings that do not pertain at all to a session. I'm thinking there needs to be a cleansing of these objects in each array that cuts them down to what we actually need (i.e. the true shape of `RSService` ([see here](https://github.com/CDCgov/prime-reportstream/pull/7923/files#diff-a72d42ba97677d060a8ea1988d66b9d9bca5cfed183eb9a12c08b17fd84d3ef4R9-R14))

Reply:
Also going to swap to a `Promise.all()` on this

Reply:
<img width="491" alt="Screen Shot 2023-01-11 at 5 14 56 PM" src="https://user-images.githubusercontent.com/4022304/211929636-20db6f33-da2c-41ac-a676-0c27fae723f3.png">
<img width="550" alt="Screen Shot 2023-01-11 at 5 15 35 PM" src="https://user-images.githubusercontent.com/4022304/211929638-3f03779f-0a3e-43a8-a710-9e92d01e7d76.png">